File: main/539-20191025.go
package main

import (
	"sort"
	"strconv"
	"strings"
)

func findMinDifference(timePoints []string) int {
	size := len(timePoints)
	min := 24 * 60
	times := make([]int, size)

	for i, timePoint := range timePoints {
		time := strings.Split(timePoint, ":")
		h, _ := strconv.Atoi(time[0])
		m, _ := strconv.Atoi(time[1])
		times[i] = h*60 + m
	}
	sort.Ints(times)

	for i := 1; i < size; i++ {
		interval := times[i] - times[i-1]
		if min > interval {
			min = interval
		}
	}
	back := (times[0] + 1440 - times[size-1])
	if min > back {
		min = back
	}
	return min
}

// func main() {
// 	timePoints := []string{"23:59", "00:00"}

// 	fmt.Println(findMinDifference(timePoints))
// }

File: main/539-20191025.rb
# @param {String[]} time_points
# @return {Integer}
def find_min_difference(time_points)
  times = []
  time_points.each do |time_point|
    h, m = time_point.split(":")
    times.push(h.to_i*60+m.to_i)
  end

  times.sort!

  min = 1440
  end_index = times.size-1

  for i in (1..end_index) do
    min = [min, times[i]-times[i-1]].min

    if i == end_index
      min = [min, 1440+times[0]-times[i]].min
    end
  end

  min
end
